packages/grafana-data/src/types/plugin.ts

```diff
@@ -110,11 +110,18 @@ export interface PluginInclude {
   path?: string;
   icon?: string;
 
-  role?: string; // "Viewer", Admin, editor???
-  addToNav?: boolean; // Show in the sidebar... only if type=page?
+  // "Admin", "Editor" or "Viewer". If set then the include will only show up in the navigation if the user has the required roles.
+  role?: string;
+
+  // Adds the "page" or "dashboard" type includes to the navigation if set to `true`.
+  addToNav?: boolean;
 
   // Angular app pages
   component?: string;
+
+  // Marks a "page" type include as a route that can be used to extend or override existing Grafana pages.
+  // A page with this flag enabled will not show up under the plugins own sub-pages, but can be used as a standalone plugin page to show up under a certain navigation section using the `navigation.app_standalone_pages` Grafana setting.
+  isCorePage?: boolean;
```

Comment:
Question: would it be possible to use the Grafana configuration to figure out if it can be used as a core page? I guess you need to specify something in the Grafana configuration to allow a plugin to override parts of the core pages?

Reply:
Basically do something like this and use the `plugin-id` from the configuration to figure this out instead of needing the plugin to specify that is should be served as a core page:

```bash
# Specify a plugin page using its `path` and define which section it should live under
[navigation.app_standalone_pages]
"/a/plugin-id/connections" = connections # section ID
```

Reply:
Yeah, that's a very fair question and I was expecting it coming. :)

In most of the cases what you are suggesting should work. **The thing becomes tricky when we are trying to use a plugin to render under a core URL (not `/a/:plugin-id/...`).** If a plugin is exposing a page like that, then that page will probably only work and make sense if it is rendered under the correct section.

Example:

**`custom.ini`**
```ini
[navigation.app_standalone_pages]
"/connections/your-connections/integrations" = connections # section ID
```

**`plugin.json`**
```javascript
// ...
"includes": [
   {
      "type": "page",
      "name": "Random page",
      "path": "/a/my-random-app/random-page",
      "role": "Admin",
      "addToNav": true,
    },
   // The following page is expected to be rendered under `/connections/your-connections/integrations`, so it can
   // seamlessly extend Grafana's core capabilities. In case it wouldn't be configured in the INI, then it would show up
   // under the plugin's section and wouldn't work, as it is expected to live under a certain core URL. 
    {
      "type": "page",
      "name": "Integrations",
      "path": "/connections/your-connections/integrations",
      "role": "Admin",
      "addToNav": true,
      "isCorePage": true
    },
  ],
```

The edge-case is that currently we enable plugins to register custom paths (apart from `/a/:pluginId`) in their includes, and these will become links. In these cases though (if a page is meant to be extending core) they don't make sense as a simple link in case the server admin doesn't configure them.

Pages that live under `/a/:pluginId` can be more easily "moved around" in the nav-tree, as they will be always routed to the `AppRootPage` on the frontend and will always be displayed under their "own url". In these cases however these will be "just links" under other sections if they are used as standalone pages, and they won't give the feeling of an integrated feature.

Does this answer your question @mckn ?

Reply:
Yeah, I think I understand. A couple of follow up questions:

- Are you always replacing a whole section? Or do you replace e.g. one page of a section?
- Can we introduce Page IDs for the pages below a section that you are allowed to replace?

What about doing something like this (I have assumed that you want to replace single pages below a section):

`**plugin.json**`

```json
{ 
   "includes": [ 
      { 
         "type": "page", 
         "name": "Integrations", 
         "path": "/connections/your-connections/integrations", 
         "role": "Admin", 
         "addToNav": false, 
         "replaceInNav": {
            "section": "connections",
            "page": "integrations",
         }
      }, 
   ] 
}
```

Potentially added by the server admin:

```bash
# [navigation.app_sections.<Section ID>]
# Format: <Page ID> = <Plugin ID>
[navigation.app_sections.connections]
integrations = grafana-connections-app
```

I'm not sure this makes sense but if we only add a new attribute and don't reply on the `addToNav` then we won't risk of ending up in the edge case that you mentioned. Also - by using page ID we still keep the flexibility to refactor paths/routes within Grafana without breaking plugins. I guess a drawback with that is that we need to expose some kind of API in plugins to be able to resolve the path to a sectionId + pageId if you want to generate links in the plugin page.

I have probably missed something but would this work?

Reply:
And in the admin config you are basically saying: "This plugin is allowed to replace the navigation page in this section"

Reply:
So after rubber ducking this a bit. Ignore my comment above. What about the following:

1. Remove the `isCorePage` property.
2. Only try to add this behavior if the `addToNav` is `false`. 
3. Open an issue with the goal to improve the `[navigation.app_standalone_pages]` configuration to make it a bit more descriptive.

Reply:
Created a follow up issue: https://github.com/grafana/grafana/issues/58336